### PR TITLE
WIRE-838 - Support for managing organization groups

### PR DESCRIPTION
**Pull request by oskarwojciski**
Add support for managing organization groups (only for organizations with rbacv2 enabled)

### CODE REVIEW

File: castai/resource_organization_group.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+	client := meta.(*ProviderConfig).api
+
+	group, err := getGroup(client, ctx, organizationID, groupID)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("getting group for read: %w", err))
+	}
+
+	err = assignGroupData(group, data)
+	if err != nil {
```
Comment (radekska) [inline castai/resource_organization_group.go:130]:
```suggestion
	if err := assignGroupData(group, data); err != nil {
```

File: castai/resource_organization_group.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+)
+
+const (
+	FieldOrganizationGroupOrganizationID = "organization_id"
+	FieldOrganizationGroupName           = "name"
+	FieldOrganizationGroupDescription    = "description"
+	FieldOrganizationGroupMembers        = "members"
+	FieldOrganizationGroupMember         = "member"
+	FieldOrganizationGroupMemberID       = "id"
+	FieldOrganizationGroupMemberEmail    = "email"
```
Comment (radekska) [inline castai/resource_organization_group.go:23]:
```suggestion
	FieldOrganizationGroupMemberEmail    = "email"
	FieldOrganizationGroupMemberKind     = "kind" # USER or SERVICE_ACCOUNT
```
Had to introduce an extra attribute to distinct whether a group member is a user or a serviceAccount as both cases are valid from the business perspective - https://api.cast.ai/v1/spec/#/RbacServiceAPI/RbacServiceAPI_CreateGroup

Comment (oskarwojciski) [reply castai/resource_organization_group.go:26]: @radekska I have added support for member kind

File: castai/resource_organization_group_test.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/stretchr/testify/require"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+	mock_sdk "github.com/castai/terraform-provider-castai/castai/sdk/mock"
+)
+
+func TestOrganizationGroupReadContext(t *testing.T) {
```
Comment (radekska) [inline castai/resource_organization_group_test.go:19]: I'd test other contexts as well 🙏

**Comment by radekska** [comment]
Does the resource support service account as members as well?
I'm trying to do the following:
```terraform
resource "castai_organization_group" "first_group" {
  organization_id = data.castai_organization.test.id
  name            = "first-group"
  description     = "A description of the first group."

  members {
    member {
      kind  = "user"
      id    = "2feb1867-c071-4a14-adcf-f6cec373a229"
      email = "radoslaw@cast.ai"
    }
    member {
      kind  = "service_account"
      id    = "25f88e1a-037f-4930-8dd6-363533514b0d"
      email = "25f88e1a-037f-4930-8dd6-363533514b0d@a21389e6-852c-40d5-99c7-1e2fc7971f7e.iam.castserviceaccount.local"
    }
  }
}
```

And getting:
```
╷
│ Error: update group: expected status code 200, received: status=400 body={"message":"invalid request", "fieldViolations":[{"field":"defi
nition.members.1", "description":"invalid member"}]}
│ 
│   with castai_organization_group.first_group,
│   on main.tf line 17, in resource "castai_organization_group" "first_group":
│   17: resource "castai_organization_group" "first_group" {
│ 
```

Org ID: `21389e6-852c-40d5-99c7-1e2fc7971f7e`
Env: `prod-master` :kek:

**Review by radekska — APPROVED** [review]

**Pull request merged**